### PR TITLE
update(panel): Support multiple panels and clean up some code.

### PR DESCRIPTION
**Pull request by KarenParker**
Fixes the problem of not resetting the defaults when there are multiple panels.

@ErinCoughlan @jelbourn @gmoothart @DerekLouie @ThomasBurleson  Please review.

### CODE REVIEW

File: src/components/panel/panel.js [hunk 1]
```diff
@@ -649,6 +652,8 @@ function MdPanelService($rootElement, $rootScope, $injector) {
 MdPanelService.prototype.create = function(opt_config) {
   var configSettings = opt_config || {};
 
+  angular.copy(this._defaultConfigOptions, this._config);
```
Comment (ErinCoughlan) [inline src/components/panel/panel.js]: nit: What about using extend for both? As written, it harder to remember that one of these copies left to right and the other right to left.

Comment (KarenParker) [inline src/components/panel/panel.js]: Extend won't copy the values, just point to the other object's values. This is what I am trying to avoid.

**Comment by ErinCoughlan** [comment]
test to make sure multiple panels is possible?

File: src/components/panel/panel.js (hunk 1 above)
Comment (jelbourn) [inline src/components/panel/panel.js]: Do you need a full deep-copy? The performance of `angular.copy` is not the best.

**Comment by KarenParker** [comment]
Test added.

File: src/components/panel/panel.js (hunk 1 above)
Comment (KarenParker) [inline src/components/panel/panel.js]: The deep-copy isn't needed anymore. Code has been updated.

**Comment by ErinCoughlan** [comment]
LGTM

**Comment by jelbourn** [comment]
LGTM

**Pull request merged**